Block/BlockContextManagerInterface.php

```diff
@@ -39,6 +39,13 @@ public function addSettingsByType($type, array $settings, $replace = false);
     public function addSettingsByClass($class, array $settings, $replace = false);
 
     /**
+     * @param string $type    Block type to check for
+     *
+     * @return bool
+     */
+    public function exists($type);
```

Comment:
adding a method here is BC break

we can merge this without this method and add a separate PR for the next major IMO

ping @Soullivaneuh